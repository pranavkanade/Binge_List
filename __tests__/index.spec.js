import React from "react";
import { shallow, mount } from "enzyme";
import MoviesInTheaters from "../src/components/index/inTheater";
import PopularMovies from "../src/components/index/popular";
import { PopularData } from "../test_data/apiData";
import renderer from "react-test-renderer";

describe("Testing Index page", () => {
  describe("Test Movies in Theaters", () => {
    it("Fetches data from server and updates the state", () => {
      const mockSuccessResponse = PopularData;
      const mockJsonPromise = Promise.resolve(mockSuccessResponse);
      const mockFetchPromise = Promise.resolve({
        json: () => mockJsonPromise
      });
      global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

      const wrapper = mount(<MoviesInTheaters />);
      expect(global.fetch).toHaveBeenCalledTimes(1);

      process.nextTick(() => {
        expect(wrapper.state()).toEqual({
          moviesNowPlaying: PopularData.results
        });
      });
    });
  });

  describe("Test Popular Movies", () => {
    it("Fetches data from server and updates the state", () => {
      const mockSuccessResponse = PopularData;
      const mockJsonPromise = Promise.resolve(mockSuccessResponse);
      const mockFetchPromise = Promise.resolve({
        json: () => mockJsonPromise
      });
      global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

      const wrapper = mount(<PopularMovies />);
      expect(global.fetch).toHaveBeenCalledTimes(1);

      process.nextTick(() => {
        expect(wrapper.state().moviesList.length).toEqual(
          PopularData.results.length
        );
      });
    });

    it("Shows only 8 movies initiallly", () => {
      const mockSuccessResponse = PopularData;
      const mockJsonPromise = Promise.resolve(mockSuccessResponse);
      const mockFetchPromise = Promise.resolve({
        json: () => mockJsonPromise
      });
      global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

      const wrapper = mount(<PopularMovies />);
      expect(wrapper.state().showCount).toEqual(8);
    });
  });
});

describe("Snapshot testing Index Page", () => {
  describe("Test Movies in Theaters", () => {
    it("Renders correctly", () => {
      const mockSuccessResponse = PopularData;
      const mockJsonPromise = Promise.resolve(mockSuccessResponse);
      const mockFetchPromise = Promise.resolve({
        json: () => mockJsonPromise
      });
      global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);

      const moviesInTheater = renderer.create(<MoviesInTheaters />).toJSON();
      expect(moviesInTheater).toMatchSnapshot();

      global.fetch.mockClear();
      delete global.fetch;
    });
  });
});
