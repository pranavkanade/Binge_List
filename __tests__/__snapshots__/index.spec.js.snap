// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Index Page Test Movies in Theaters Renders correctly 1`] = `
Array [
  <h2
    className="ui left aligned header"
  >
    Available in Theaters Now!
    <button
      className="ui orange basic right floated button"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Show All
    </button>
  </h2>,
  <div
    className="ui basic segment"
  >
    <div
      className="ui basic segment"
    >
      <div
        className="ui active transition visible inverted dimmer"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <div
            className="ui massive text loader"
          >
            Loading
          </div>
        </div>
      </div>
      <div
        className="ui fluid placeholder"
        style={
          Object {
            "height": 350,
          }
        }
      >
        <div
          className="image"
        />
      </div>
    </div>
  </div>,
]
`;
